### PR TITLE
Add documentation for usage of spack-execute module

### PR DESCRIPTION
**Pull request by nick-stroud**
Description in title.

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: community/modules/scripts/spack-execute/README.md [hunk 1]
```diff
@@ -1,3 +1,78 @@
+## Description
+
+This module creates a script that defines a build using Spack and perform any
```
Comment (douglasjacobsen) [inline community/modules/scripts/spack-execute/README.md:3]:
```suggestion
This module creates a script that defines a software build using Spack and performs any
```

File: community/modules/scripts/spack-execute/README.md [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+additional customization to a Spack installation.
+
+There are two main variable inputs that can be used to define a Spack build:
+`data_files` and `commands`.
+
+- `data_files`: Any files specified will be transferred to the machine running
+  outputted script. Data file `content` can be defined can be defined inline in
```
Comment (douglasjacobsen) [inline community/modules/scripts/spack-execute/README.md:10]:
```suggestion
  outputted script. Data file `content` can be defined inline in
```

File: community/modules/scripts/spack-execute/README.md [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+  the blueprint or can point to a `source`, an absolute local path of a file.
+  This can be used to transfer environment definition files, config definition
+  files, GPG keys, or software licenses. `data_files` are transferred before
+  `commands` are run.
+- `commands`: A script that is run. This can be used to perform actions such as
+  install compilers & packages, create environments, add a build cache, and
```
Comment (douglasjacobsen) [inline community/modules/scripts/spack-execute/README.md:16]:
```suggestion
  installation of compilers & packages, environment creation, adding a build cache, and
```

File: community/modules/scripts/spack-execute/README.md [hunk 4]
```diff
@@ -1,3 +1,78 @@
...
+  modify the spack configuration.
```
Comment (douglasjacobsen) [inline community/modules/scripts/spack-execute/README.md:17]:
```suggestion
  modifying the spack configuration.
```

**Review by douglasjacobsen — APPROVED** [review]
Looks good. A few suggested changes, but otherwise I think LGTM. Feel free to accept or ignore as you  please.

File: community/modules/scripts/spack-execute/README.md (hunk 1 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-execute/README.md]: Done

File: community/modules/scripts/spack-execute/README.md (hunk 2 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-execute/README.md]: Done

File: community/modules/scripts/spack-execute/README.md (hunk 3 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-execute/README.md]: Done

File: community/modules/scripts/spack-execute/README.md (hunk 4 above)
Comment (nick-stroud) [reply community/modules/scripts/spack-execute/README.md]: Done

**Pull request merged**